### DISCUSSION TITLE
What;s the purpose of importing ruff crates directly as dependencies?

### DISCUSSION THREAD

**Discussion by clin1234** [gauge-sh/tach#168] [General]
Title

**Comment by emdoyle** [comment]
The main reason is that their recursive descent parser for Python ASTs was faster and more robust than the RustPython parser. They don't publish on crates.io (at least not when I added the dep) so we just use them as a git dependency through cargo.

More details here: https://www.gauge.sh/blog/parsing-python-asts-20x-faster-with-rust
